testDocValuesFieldExistsForKeyword fails

**Build scan**: https://gradle-enterprise.elastic.co/s/2oforrhohk5iw/console-log/raw

**Repro line**:

```console
./gradlew ':server:test' --tests \
"org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testDocValuesFieldExistsForKeyword" \
-Dtests.seed=A1D2B5023CCC99A3 -Dtests.locale=it-IT -Dtests.timezone=Asia/Ashkhabad -Druntime.java=8
```
**Reproduces locally?**: YES

**Applicable branches**: 7.x, master

```
org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests > testDocValuesFieldExistsForKeyword FAILED
    java.lang.AssertionError: 
    Expected: <0>
         but: was <1>
        at __randomizedtesting.SeedInfo.seed([A1D2B5023CCC99A3:A3C1EA12F1924877]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.lambda$docValuesFieldExistsTestCase$45(FiltersAggregatorTests.java:954)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:639)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.lambda$debugTestCase$9(AggregatorTestCase.java:593)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.withIndex(AggregatorTestCase.java:557)
        at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:593)
        at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.docValuesFieldExistsTestCase(FiltersAggregatorTests.java:941)
        at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testDocValuesFieldExistsForKeyword(FiltersAggregatorTests.java:909)
```

A slightly different failure in master, reproducible with
```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testDocValuesFieldExistsForKeyword" -Dtests.seed=23090D7660128ECE -Dtests.locale=nl-NL -Dtests.timezone=America/North_Dakota/Beulah -Druntime.java=16
```
The failure: 
```
java.lang.AssertionError: 
Expected: <0>
     but: was <2>
	at __randomizedtesting.SeedInfo.seed([23090D7660128ECE:211A5266AD4C5F1A]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.lambda$docValuesFieldExistsTestCase$46(FiltersAggregatorTests.java:911)
	at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:633)
	at org.elasticsearch.search.aggregations.AggregatorTestCase.lambda$debugTestCase$9(AggregatorTestCase.java:587)
	at org.elasticsearch.search.aggregations.AggregatorTestCase.withIndex(AggregatorTestCase.java:551)
	at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:587)
	at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.docValuesFieldExistsTestCase(FiltersAggregatorTests.java:898)
	at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testDocValuesFieldExistsForKeyword(FiltersAggregatorTests.java:871)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

--- a/server/src/test/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregatorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregatorTests.java
@@ -885,20 +885,16 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
         IntFunction<List<? extends IndexableField>> buildDocWithField
     ) throws IOException {
         AggregationBuilder builder = new FiltersAggregationBuilder("test", new KeyedFilter("q1", exists));
-        CheckedConsumer<RandomIndexWriter, IOException> buildIndex = iw -> {
-            for (int i = 0; i < 10; i++) {
-                iw.addDocument(buildDocWithField.apply(i));
-            }
-            for (int i = 0; i < 10; i++) {
-                iw.addDocument(List.of());
-            }
-        };
         // Exists queries convert to MatchNone if this isn't defined
         FieldNamesFieldMapper.FieldNamesFieldType fnft = new FieldNamesFieldMapper.FieldNamesFieldType(true);
         debugTestCase(
             builder,
             new MatchAllDocsQuery(),
-            buildIndex,
+            iw -> {
+                for (int i = 0; i < 10; i++) {
+                    iw.addDocument(buildDocWithField.apply(i));
+                }
+            },
             (InternalFilters result, Class<? extends Aggregator> impl, Map<String, Map<String, Object>> debug) -> {
                 assertThat(result.getBuckets(), hasSize(1));
                 assertThat(result.getBucketByKey("q1").getDocCount(), equalTo(10L));
@@ -913,7 +909,14 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
             fieldType,
             fnft
         );
-        withAggregator(builder, new MatchAllDocsQuery(), buildIndex, (searcher, aggregator) -> {
+        withAggregator(builder, new MatchAllDocsQuery(), iw -> {
+            for (int i = 0; i < 10; i++) {
+                iw.addDocument(buildDocWithField.apply(i));
+            }
+            for (int i = 0; i < 10; i++) {
+                iw.addDocument(List.of());
+            }
+        }, (searcher, aggregator) -> {
             long estimatedCost = ((FiltersAggregator.FilterByFilter) aggregator).estimateCost(Long.MAX_VALUE);
             Map<String, Object> debug = new HashMap<>();
             aggregator.collectDebugInfo(debug::put);

